src/dense.jl

```diff
@@ -111,21 +109,14 @@ end
 mode(::AutoFiniteDiff) = ForwardMode()
 
 function Base.show(io::IO, backend::AutoFiniteDiff)
-    s = "AutoFiniteDiff("
-    if backend.fdtype != Val(:forward)
-        s *= "fdtype=$(repr(backend.fdtype, context=io)), "
-    end
-    if backend.fdjtype != backend.fdtype
-        s *= "fdjtype=$(repr(backend.fdjtype, context=io)), "
-    end
-    if backend.fdhtype != Val(:hcentral)
-        s *= "fdhtype=$(repr(backend.fdhtype, context=io)), "
-    end
-    if endswith(s, ", ")
-        s = s[1:(end - 2)]
-    end
-    s *= ")"
-    print(io, s)
+    print(io, AutoFiniteDiff, "(")
+    backend.fdtype != Val(:forward) &&
+        print(io, "fdtype=", repr(backend.fdtype; context = io), ", ")
+    backend.fdjtype != backend.fdtype &&
+        print(io, "fdjtype=", repr(backend.fdjtype; context = io), ", ")
+    backend.fdhtype != Val(:hcentral) &&
+        print(io, "fdhtype=", repr(backend.fdhtype; context = io))
```

Comment:
These don't show the Val?

Reply:
Yes they do, there's just no `Val` in the test case I displayed above. Take a look here:

```julia
julia> string(AutoFiniteDiff(fdjtype=Val(:whatever)))
"AutoFiniteDiff(fdjtype=Val{:whatever}(), )"
```

Reply:
@ChrisRackauckas I think this one is a strict improvement over the current state, what do you say?

Reply:
LGTM